src/transport/SessionManager.h

```diff
@@ -253,11 +253,11 @@ class DLL_EXPORT SessionManager : public TransportMgrDelegate
      */
     void OnMessageReceived(const Transport::PeerAddress & source, System::PacketBufferHandle && msgBuf) override;
 
-    Optional<SessionHandle> CreateUnauthenticatedSession(const Transport::PeerAddress & peerAddress,
+    Optional<SessionHandle> CreateUnauthenticatedSession(const Transport::PeerAddress & peerAddress, System::Clock::Timestamp now,
```

Comment:
A question: Since all callers pass in the exact same thing, why don't we just do the "get current monotonic timestamp" bit inside this method?  Or even further down in the session code.

Is that just to allow shimming the time source for tests?

Ideally we would find the right boundary where we can avoid having all consumers do this by hand... For example, perhaps put this arg last and default it to the normal monotonic timestamp but that still allows other callers like tests to override?  Similar throughout.

Reply:
Yes, it is the unit-tests preventing me doing this. Let me check if I can mock system clock.